src/lib.rs

```diff
@@ -29,8 +33,22 @@ mod tests {
         assert_eq!(upload_result.height.unwrap(), 640);
         assert_eq!(upload_result.width.unwrap(), 640);
 
+        let detail_result = imagekit.get_file_details(&upload_result.file_id).await;
+        assert!(detail_result.is_ok());
+
         let delete_result = imagekit.delete(upload_result.file_id).await;
 
         assert!(delete_result.is_ok());
     }
+    #[tokio::test]
+    async fn uploads_and_retrieve_information() {
+        let imagekit = ImageKit::from_env().unwrap();
+        let file = File::open("assets/ferris.jpeg").await.unwrap();
+        let upload_file = UploadFile::from(file);
+        let opts = Options::new(upload_file, "ferris");
+        let upload_result = imagekit.upload(opts).await.unwrap();
+
+        let detail_result = imagekit.get_file_details(&upload_result.file_id).await;
+        assert!(detail_result.is_ok());
```

Comment:
I think we should check certain predictable image details to make sure data comes as expected.

Reply:
Perfect, I'll get a template of a response (image details) and try to test against it.